tests/helper.py

```diff
@@ -33,7 +33,7 @@ def assert_dtype_allclose(
 
     list_64bit_types = [numpy.float64, numpy.complex128]
     is_inexact = lambda x: dpnp.issubdtype(x.dtype, dpnp.inexact)
-    if is_inexact(dpnp_arr) or is_inexact(numpy_arr):
+    if is_inexact(dpnp_arr) and is_inexact(numpy_arr):
```

Comment:
What is the reason for change? I assumed we can use `assert_array_equal` only when both input are not of inexact dtype.

Reply:
In calculating `tol`, we use `finfo` for both of them which is only defined for `inexact` data types.